src/light-godwoken/lightGodwoken.ts

```diff
@@ -342,6 +350,35 @@ export default abstract class DefaultLightGodwoken implements LightGodwokenBase
         if (collectedSudtAmount.gte(neededSudtAmount)) break;
       }
     }
+    // if ckb is not enough, try find some free capacity from sudt cell
+    const freeCapacityProviderCells: Cell[] = [];
+    if (collectedCapatity.lt(neededCapacity)) {
+      const freeCkbCollector = this.provider.ckbIndexer.collector({
+        lock: helpers.parseAddress(this.provider.l1Address),
+        type: {
+          code_hash: this.getConfig().layer1Config.SCRIPTS.sudt.code_hash,
+          hash_type: this.getConfig().layer1Config.SCRIPTS.sudt.hash_type,
+          args: "0x",
+        },
+      });
+      for await (const cell of freeCkbCollector.collect()) {
+        const haveFreeCapacity = BI.from(SUDT_CELL_CAPACITY).lt(cell.cell_output.capacity);
+        const alreadyCollected = collectedCells.some((collectedCell) => {
+          if (
+            isEqual(collectedCell.out_point?.tx_hash, cell.out_point?.tx_hash) &&
+            isEqual(collectedCell.out_point?.index, cell.out_point?.index)
+          ) {
+            return true;
+          }
+          return false;
+        });
+        // envolve SUDT cells that has more capacity than SUDT_CELL_CAPACITY
+        if (haveFreeCapacity && !alreadyCollected) {
+          freeCapacityProviderCells.push(cell);
+          collectedCapatity = collectedCapatity.add(cell.cell_output.capacity).sub(SUDT_CELL_CAPACITY);
+        }
```

Comment:
```suggestion
if(collectedCapatity.gte(neededCapacity)) {
  break;
 }
```

can we break the loop if capacity is enough?